helusers/jwt.py

```diff
@@ -24,13 +28,24 @@ def validate(self, keys, audience):
         and validates the claims, raising an exception if anything fails."""
 
         options = {
-            "require_aud": True,
+            "verify_aud": False,
             "require_exp": True,
         }
 
-        jwt.decode(
-            self._encoded_jwt, keys, options=options, audience=audience
-        )
+        jwt.decode(self._encoded_jwt, keys, options=options)
+
+        claims = self.claims
+
+        if "aud" not in claims:
+            raise ValidationError("Missing required 'aud' claim.")
+
+        claim_audiences = claims["aud"]
+        if isinstance(claim_audiences, str):
+            claim_audiences = {claim_audiences}
+        if isinstance(audience, str):
+            audience = {audience}
+        if len(set(audience).intersection(claim_audiences)) == 0:
```

Comment:
`if not set(audience) & claim_audiences:`
maybe?

Reply:
I considered that, but went with the more verbose version. To me it felt more readable. I guess it depends on the reader what is more readable...

Reply:
Both are ok. I view set operations as analogous to basic arithmetic etc. so using native operations seems clearer